java/arcs/android/host/parcelables/ParcelableParticleSpec.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.host.parcelables
+
+import android.os.Parcel
+import android.os.Parcelable
+import arcs.core.data.ParticleSpec
+
+/** [Parcelable] variant of [ParticleSpec]. */
+data class ParcelableParticleSpec(
+    override val actual: ParticleSpec
+) : ActualParcelable<ParticleSpec> {
+    override fun writeToParcel(parcel: Parcel, flags: Int) {
+        parcel.writeString(actual.particleName)
+        parcel.writeString(actual.location)
+        parcel.writeMap(
+            actual.handles.map {
+                it.key to ParcelableHandleConnectionSpec(it.value)
+            }.toMap()
```

Comment:
Nonblocking nit: could use `associateBy` or one of the related methods which convert lists to maps.

Reply:
handles isn't a list, it's already a map. I switched it to mapValues()